### ISSUE TITLE
Hiding option group name in help?

### ISSUE THREAD

**Issue by JOJ0** [click-contrib/click-option-group#22]
Hi,
is it possible to hide the name of the option group entirely in help output. In my case I just want to make sure that _any_ of the possible options is used, otherwise error, thus I used the RequiredAnyOptionGroup.

```
$ synadm user modify userid -h
Usage: synadm user modify [OPTIONS] USER_ID

  create or modify a local user. Provide matrix user ID (@user:server) as
  argument.

Options:
  How to make this invisible?: [required_any]
    -p, --password-prompt         set password interactively.
    -P, --password TEXT           set password on command line.
    -n, --display-name TEXT       set display name. defaults to the value of
                                  user_id
...
```

Found out that leaving option group name empty string leads to this, which is nice but in my case also not what I want. I would rather prefer to leave out the rather longish (xx|yy|..) line entirely:
```
$ synadm user modify userid -h
Usage: synadm user modify [OPTIONS] USER_ID

  create or modify a local user. Provide matrix user ID (@user:server) as
  argument.

Options:
  (password_prompt|password|display_name|threepid|avatar_url|admin|deactivation_state): [required_any]
...
```

Also in this output I think it's overkill to state the list of possible/required options twice:
```
$ synadm user modify userid 
Usage: synadm user modify [OPTIONS] USER_ID
Try 'synadm user modify -h' for help.

Error: Missing one of the required options from "(password_prompt|password|display_name|threepid|avatar_url|admin|deactivation_state)" option group:
  '--password-prompt' / '-p'
  '--password' / '-P'
  '--display-name' / '-n'
  '--threepid' / '-t'
  '--avatar-url' / '-v'
  '--admin' / '-a'
  '--activate'
```

And a last question: Is it possible to change or remove the text in [brackets], eg [required_any] or [mutually_exclusive]? Sometimes it is not important to state which type of group it is - the user shouldn't bother or it should be made obvious by choosing a good name for the option group :-)

This is why: I tried to work around the issue described above with a proper option group name and realised that it doesn't really work either. If just the [brackets] could be removed it would be doable:
```
$ synadm user modify userid -h
Usage: synadm user modify [OPTIONS] USER_ID

  create or modify a local user. Provide matrix user ID (@user:server) as
  argument.

Options:
  At least one of the following options is required: [required_any]
    -p, --password-prompt         set password interactively.
    -P, --password TEXT           set password on command line.
    -n, --display-name TEXT       set display name. defaults to the value of
                                  user_id
...
```

Also with this workaround approach, this output's error message doesn't sound right then:
```
synadm user modify id
Usage: synadm user modify [OPTIONS] USER_ID
Try 'synadm user modify -h' for help.

Error: Missing one of the required options from "At least one of the following options is required" option group:
  '--password-prompt' / '-p'
  '--password' / '-P'
  '--display-name' / '-n'
  '--threepid' / '-t'
  '--avatar-url' / '-v'
  '--admin' / '-a'
  '--activate'
```

Making this error text configurable could solve my issue. Another improvement suggestion would be to just have the option to leave out the option group name part here as well, so it would just state: "Error: Missing one of the required options: "

**Comment by espdev** [comment]
Hi @JOJ0,

Thanks for feedback and report.

I think we can change code to support empty group names. It doesn't look complicated. We can just do not use/show default group name with option names enumeration. And we can hide the text in square brackets also.

**Comment by espdev** [comment]
@JOJ0, 

Actually, right now you can create a subclass of `OptionGroup` class and override `get_help_record` method to hide the group name in help:

something like this:

```python
class MyRequiredAnyGroup(RequiredAnyOptionGroup):
    def get_help_record(self, ctx):
        return None
```

However, it will not affect error messages.

Currently, I'm working on a slightly changed error messages and group name logic. Probably it will be in 0.5.2 release.

**Comment by espdev** [comment]
@JOJ0 see the new [release](https://github.com/click-contrib/click-option-group/releases/tag/v0.5.2)

Now you can do not specify a group name and use your subclass of `RequiredAnyOptionGroup` with overrided `name_extra` property with empty "name_extra" attributes:

```python
class MyRequiredAnyGroup(RequiredAnyOptionGroup):
    @property
    def name_extra(self):
        return []
```

Thus, the group name will not be displayed (also in error messages).

**Comment by JOJ0** [comment]
Hi again,
I never said thanks for this very helpful feature. Have to admit that only just now I found the time to really put it to use.

What I also find very nice is that this code doesn't break anything if someone still has 0.5.1 installed. That's great!
```
class UserModifyOptionGroup(RequiredAnyOptionGroup):
    @property
    def name_extra(self):
        return []
```

Also I played around a little and found out that I even can override get_help_record to customize to entirely customize the text below `Options:` Didn't use it in the end but very nice too!

So, thanks again! All very helpful!
Jojo
